test/pytest/test_gRPC_inference_api.py

```diff
@@ -25,66 +26,138 @@ def __get_change(current, previous):
-        ))
+        response = managment_stub.RegisterModel(
+            management_pb2.RegisterModelRequest(
+                url=path,
+                initial_workers=item["worker"],
+                synchronous=bool(item["synchronous"]),
+                model_name=item["model_name"],
+            )
+        )
 
         print(response.msg)
 
-        model_input = os.path.join(os.path.dirname(__file__), "..", item['file'])
-        prediction = __infer(test_gRPC_utils.get_inference_stub(), item['model_name'], model_input)
+        model_input = os.path.join(os.path.dirname(__file__), "..", item["file"])
+        prediction = __infer(
+            test_gRPC_utils.get_inference_stub(), item["model_name"], model_input
+        )
 
         print("Prediction is : ", str(prediction))
 
-        if 'expected' in item:
+        if "expected" in item:
             try:
                 prediction = literal_eval(prediction)
             except SyntaxError:
                 pass
 
-            if isinstance(prediction, list) and 'tolerance' in item:
-                assert len(prediction) == len(item['expected'])
+            if isinstance(prediction, list) and "tolerance" in item:
+                assert len(prediction) == len(item["expected"])
                 for i in range(len(prediction)):
-                    assert __get_change(prediction[i], item['expected'][i]) < item['tolerance']
-            elif isinstance(prediction, dict) and 'tolerance' in item:
-                assert len(prediction) == len(item['expected'])
+                    assert (
+                        __get_change(prediction[i], item["expected"][i])
+                        < item["tolerance"]
+                    )
+            elif isinstance(prediction, dict) and "tolerance" in item:
+                assert len(prediction) == len(item["expected"])
                 for key in prediction:
-                    assert __get_change(prediction[key], item['expected'][key]) < item['tolerance']
+                    assert (
+                        __get_change(prediction[key], item["expected"][key])
+                        < item["tolerance"]
+                    )
             else:
-                assert str(prediction) == str(item['expected'])
+                assert str(prediction) == str(item["expected"])
+
+        response = managment_stub.UnregisterModel(
+            management_pb2.UnregisterModelRequest(
+                model_name=item["model_name"],
+            )
+        )
+
+        print(response.msg)
+
+
+def __infer_stream(stub, model_name, model_input):
+    with open(model_input, "rb") as f:
+        data = f.read()
+
+    input_data = {"data": data}
+    responses = stub.StreamPredictions(
+        inference_pb2.PredictionsRequest(model_name=model_name, input=input_data)
+    )
+
+    prediction = []
+    for resp in responses:
+        prediction.append(resp.prediction.decode("utf-8"))
+
+    return " ".join(prediction)
```

Comment:
will this be a list of partial predictions so a single prediction or a list of multiplee predictions with batch size 1

Reply:
prediction is a list of partial prediction responses. Here returns joining all of the partial response together to make the later comparing with expected values much easier.